docs/content/docs/test/Toast.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Testing toasts
+type: docs
+---
+
+# Testing toasts
+
+[android.widget.Toast](https://developer.android.com/reference/android/widget/Toast)
+
+{{< hint warning>}}
+Testing toasts reliably is a hard task because some other toast can show up and yours will be queued. 
+We hack app with proxy object to mitigate this.
+{{< /hint >}}
+
+{{< hint warning>}}
+It is also recommended to use [snackbar](https://developer.android.com/reference/com/google/android/material/snackbar/Snackbar), 
+or your custom way to display brief unintrusive messages instead. So you should consider not testing it at all.
+{{< /hint >}}
+
+## Proxy toast
+
+Use provided extension methods to be able to spy on `showToast` functions.
+
+```kotlin
+import com.avito.android.util.showToast
+
+showToast("I'am a toast!")
+```
+
+```kotlin
+implementation("com.avito.android:proxy-toast:$version")
+```
+
+## ToastRule
+
+ToastRule changes proxy toast's singleton before the test to store `showToast` invocations.
```

Comment:
Change abstract `changes`
```suggestion
ToastRule clear toasts history of proxy toast's singleton before the test.
```

Reply:
i don't understand the phrase now

Reply:
I've updated

Reply:
@eugene-krivobokov , @int02h, what do you think? which variant is better?

Reply:
it is not necessary to clear collection before test as we run all of them in isolation right now;
i don't want to focus on it here.
All i wanted to say that Rule changes singleton value to it's own Proxy object and keeps tracking of invocations. 
I see my original explanation good enough

Reply:
>  Rule changes singleton value to it's own Proxy object and keeps tracking of invocations.

The more I read this the more I think that this is excessive implementation details.

Reply:
What is a value of toatrule info? Implementation details that teach how to use rule correctly give a value.